Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -2395,23 +2345,21 @@ namespace Babylon
                 layer.Set("framebufferHeight", Napi::Value::From(env, HEIGHT));
             }
 
-            FrameBufferData* GetFrameBufferForEye(const std::string& eye) const
+            Napi::Value GetRenderTargetForEye(const std::string& eye) const
             {
-                FrameBufferData* frameBufferData = m_xr.ActiveFrameBuffers()[XREye::EyeToIndex(eye)];
-                frameBufferData->OwnedByJS = true;
-                return frameBufferData;
+                return m_xr->GetRenderTargetForViewIndex(XREye::EyeToIndex(eye));
             }
 
-            xr::Size GetWidthAndHeightForViewIndex(size_t viewIndex) const
+            void SetRenderTextureFunctions(const Napi::Function& createFunction, const Napi::Function& destroyFunction)
             {
-                return m_xr.GetWidthAndHeightForViewIndex(viewIndex);
+                return m_xr->SetRenderTextureFunctions(createFunction, destroyFunction);
             }
 
         private:
-            NativeXr m_xr;
+            JsRuntimeScheduler m_runtimeScheduler;
+            std::shared_ptr<NativeXr> m_xr;
```

Comment:
I searched around, but didn't see; does anyone ever share ownership of this? If not, could it go back to not being a `shared_ptr`?

Reply:
look for thisRef which is captured everywhere inside itself.